### PR TITLE
CVE-2009-2693

### PR DESCRIPTION
**Pull request by dxc7790**
Added information for CVE-2009-2693

### CODE REVIEW

**Comment by dxc7790** [comment]
The Ci output is telling me something is blank when it's not. I'm not fixing it

File: CVE-2009-2693.yml [hunk 1]
```diff
@@ -0,0 +1,220 @@
+CVE: CVE-2009-2693
+
```
Comment (mesternefeld) [inline CVE-2009-2693.yml:2]: Please delete this file. It should be in the other directory.

Comment (dxc7790) [reply CVE-2009-2693.yml:2]: I don't know how this file got here but thank you for noticing it. I'll delete it.

File: cves/CVE-2009-2693.yml [hunk 2]
```diff
@@ -1,205 +1,215 @@
...
+  integrity checks on this file to make sure you fill everything out properly.
+  If you are a student, we cannot accept your work as finished unless curated is
+  set to true.
+curated: true
+
+reported_instructions: |
+  Was there a date that this vulnerability was reported to the team? You can
+  find this in changelogs, blogs, bug reports, or perhaps the CVE data.
+  Please enter your date in YYYY-MM-DD format.
+reported:
```
Comment (bjt6087) [inline cves/CVE-2009-2693.yml]: On the tomcat website "This was first reported to the Tomcat security team on 30 Jul 2009 and made public on 1 Mar 2010."

File: cves/CVE-2009-2693.yml [hunk 3]
```diff
@@ -1,205 +1,215 @@
...
+    Maintainability? Requirements? Miscommunications?
+    Look at the CWE entry for this vulnerability and examine the mitigations
+    they have written there. Are they doing those? Does the fix look proper?
+    Use those questions to inspire your answer. Don't feel obligated to answer
+    every one. Write a thoughtful entry here that those ing the software
+    engineering industry would find interesting.
+  answer: The mistake that was made here was simple. The developer neglected to consider
+    that someone might try to navigate outside of the root directory and never
+    put in test cases and exceptions for when that happens. It isn't exactly a
+    mistake but more of an oversight on the developers part.
```
Comment (bjt6087) [inline cves/CVE-2009-2693.yml]: What did they do to fix this vulnerability? Was the fix that was implemented adequate?

**Comment by bjt6087** [comment]
Nice work! 2 upvotes

File: cves/CVE-2009-2693.yml (hunk 2 above)
Comment (dxc7790) [reply cves/CVE-2009-2693.yml]: Thank you, I must have missed this. I made the fix in the latest commit.

File: cves/CVE-2009-2693.yml (hunk 3 above)
Comment (dxc7790) [reply cves/CVE-2009-2693.yml]: The developer added checks for path traversals in the section that was creating paths. They added explicit checks for "../" in the path. I added this to the most recent commit.

File: cves/CVE-2009-2693.yml [hunk 4]
```diff
@@ -1,205 +1,216 @@
...
+      fill in the name there.
+    * The "automated" flag can be true, false, or nil.
+    If there is no evidence as to how this vulnerability was found, then you
+    may leave the entries blank except for "answer", BUT please write down
+    where you looked in "answer".
+  answer: The SVN page for the commit does not indicate how the vulnerability was found
+    https://svn.apache.org/repos/asf/tomcat/trunk@892795
+    and the links within the CVE page also do not show how this was discovered
+    https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2009-2693
+  date:
```
Comment (olucomedy) [inline cves/CVE-2009-2693.yml]: Could you fill in the date when this vulnerability discovered?

File: cves/CVE-2009-2693.yml [hunk 5]
```diff
@@ -1,205 +1,216 @@
...
+  automated:
```
Comment (SnowBirdMV) [inline cves/CVE-2009-2693.yml]: Automated field is blank?

File: cves/CVE-2009-2693.yml [hunk 6]
```diff
@@ -1,205 +1,216 @@
...
+  question: |
+    In your opinion, after all of this research, what mistakes were made that
+    led to this vulnerability? Coding mistakes? Design mistakes?
+    Maintainability? Requirements? Miscommunications?
+    Look at the CWE entry for this vulnerability and examine the mitigations
+    they have written there. Are they doing those? Does the fix look proper?
+    Use those questions to inspire your answer. Don't feel obligated to answer
+    every one. Write a thoughtful entry here that those ing the software
+    engineering industry would find interesting.
+  answer: The mistake that was made here was simple. The developer neglected to consider
```
Comment (SnowBirdMV) [inline cves/CVE-2009-2693.yml:212]: may want to mention the lack of unit tests around this prior to the vulnerability being discovered.

**Comment by SnowBirdMV** [comment]
Good job with the descriptions, +3 upvotes

File: cves/CVE-2009-2693.yml (hunk 4 above)
Comment (dxc7790) [reply cves/CVE-2009-2693.yml]: I must have missed this field, thanks for letting me know!

File: cves/CVE-2009-2693.yml (hunk 5 above)
Comment (dxc7790) [reply cves/CVE-2009-2693.yml]: I guess I forgot to fill this one in. Thanks for letting me know!

File: cves/CVE-2009-2693.yml (hunk 6 above)
Comment (dxc7790) [reply cves/CVE-2009-2693.yml:212]: I added a mention of lack of unit testing. Good idea!

**Comment by exl5325** [comment]
2 upvotes

**Comment by allisonsuarez** [comment]
+1 +1

File: cves/CVE-2009-2693.yml [hunk 7]
```diff
@@ -1,205 +1,218 @@
...
+  descriptions are a fine start, but they can be kind of jargony.
+  Rewrite this description in your own words. Make it interesting and easy to
+  read to anyone with some programming experience. We can always pull up the NVD
+  description later to get more technical.
+  Try to still be specific in your description, but remove Chromium-specific
+  stuff. Remove references to versions, specific filenames, and other jargon
+  that outsiders to Chromium would not understand. Technology like "regular
+  expressions" is fine, and security phrases like "invalid write" are fine to
+  keep too.
+description: The vulnerability allows for an attacker to create, modify, or delete files
```
Comment (MaeveMonster) [inline cves/CVE-2009-2693.yml:46]: This is an excellent description. Perfect amount of detail and good use of active voice.

File: cves/CVE-2009-2693.yml [hunk 8]
```diff
@@ -1,205 +1,218 @@
...
+  - commit: none
+    note: The fix worked
+
+upvotes_instructions: |
+  Students: when initially writing this, ignore this upvotes number.
+  Once this work is being reviewed, you will be giving a certain amount of
+  upvotes to each vulnerability you see. Your peers will tell you how
+  interesting they think this vulnerability is, and you'll add that to the
+  upvotes score on your branch.
+upvotes: 5
```
Comment (MaeveMonster) [inline cves/CVE-2009-2693.yml:90]: add 4 upvotes from me

**Pull request merged**